subprojects/test-runner/test-inhouse-runner/src/main/kotlin/com/avito/android/rule/InHouseScenarioScreenRule.kt

```diff
@@ -0,0 +1,64 @@
+@file:Suppress("unused")
+package com.avito.android.rule
+
+import android.app.Activity
+import android.app.Instrumentation
+import android.content.Context
+import android.content.Intent
+import androidx.lifecycle.Lifecycle
+import androidx.test.core.app.ActivityScenario
+import androidx.test.platform.app.InstrumentationRegistry
+import com.avito.android.util.waitForAssertion
+import com.google.common.truth.Truth.assertThat
+import com.google.common.truth.Truth.assertWithMessage
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+
+abstract class InHouseScenarioScreenRule<A : Activity>(activityClass: Class<A>) : TestRule {
+
+    protected val androidInstrumentation: Instrumentation = InstrumentationRegistry.getInstrumentation()
+    protected val appContext: Context = androidInstrumentation.targetContext.applicationContext
+    protected val testContext: Context = androidInstrumentation.context
+
+    private val activityRule = ActivityScenarioRule(activityClass)
+
+    val checks = ChecksLibrary { activityRule.scenario }
+
+    val activityResult: Instrumentation.ActivityResult
+        get() = activityRule.scenario.result
+
+    val scenario: ActivityScenario<A>
+        get() = activityRule.scenario
+
+    fun launchActivity(startIntent: Intent?): ActivityScenario<A> = activityRule.launchActivity(startIntent)
+
+    class ChecksLibrary<A : Activity>(private val scenarioFunc: () -> ActivityScenario<A>) {
+
+        /**
+         * Asserts that activity has been closed and destroyed.
+         * For more information on lifecycle states, see [androidx.lifecycle.Lifecycle.State]
+         */
+        fun isDestroyed() {
+            waitForAssertion { assertThat(scenarioFunc().state).isEqualTo(Lifecycle.State.DESTROYED) }
+        }
+
+        /**
+         * Asserts that activity is currently alive.
+         * For more information on lifecycle states, see [androidx.lifecycle.Lifecycle.State]
+         */
+        fun isNotDestroyed() {
+            waitForAssertion { assertThat(scenarioFunc().state).isNotEqualTo(Lifecycle.State.DESTROYED) }
+        }
+
```

Comment:
Will clients need to check other states?
What about more generic way?

```kotlin
isInState(Lifecycle.State.RESUMED)
```

Reply:
Looks good to me. Replaced with 3 new methods